go/private/rules/wrappers.bzl

```diff
@@ -61,6 +66,45 @@ def go_binary_macro(name, **kwargs):
   _deprecate_importpath(name, "go_binary", kwargs)
   _cgo(name, kwargs)
   go_binary(name = name, **kwargs)
+  _go_binary_c_archive_shared(name, kwargs)
+
+def _go_binary_c_archive_shared(name, kwargs):
```

Comment:
FYI, again to echo my comment about the C++ sandwich above, I think this interface is fine for now, but it may need to change in the future.

Reply:
i'd love to! but for now it'll have to do i think